### PR TITLE
Process generation

### PR DESCRIPTION
**Pull request by bgoesswe**
Dynamic process generation. 
See #40 for more information.

### CODE REVIEW

File: openeo/internal/filter_bbox.json [hunk 1]
```diff
@@ -0,0 +1,123 @@
+{
```
Comment (soxofaan) [inline openeo/internal/filter_bbox.json]: I think we should avoid mirroring json files from openeo-api and -processes

File: openeo/internal/processes.py [hunk 2]
```diff
@@ -0,0 +1,902 @@
+def ln(cube, x):
```
Comment (soxofaan) [inline openeo/internal/processes.py]: I would also add some metadata in a comment in the header of this file with information like:
- which backend is this from
- at what time was it generated

File: openeo/internal/processes.py [hunk 3]
```diff
@@ -0,0 +1,902 @@
...
+    """The natural logarithm is the logarithm to the base *e* of the number `x`. This
+process is an alias for the *log* process with the base set to *e*: `log(x,
```
Comment (soxofaan) [inline openeo/internal/processes.py]: FYI: the `textwrap` module has an `indent` function (https://docs.python.org/3/library/textwrap.html#textwrap.indent)  with which you can make the indentation of these docblocks nicer

**Comment by soxofaan** [comment]
FYI: regarding the notes about the example scripts: we can delegate clean up also to #111

File: examples/eurac_example.py [hunk 4]
```diff
@@ -1,43 +1,56 @@
 import openeo
 import logging
 import json
+from openeo.rest.imagecollectionclient import ImageCollectionClient
```
[the remark this answers is not in the document]
Comment (bgoesswe) [reply examples/eurac_example.py]: fixed in the last commit

File: examples/eurac_example.py [hunk 5]
```diff
@@ -1,43 +1,56 @@
...
 print(cap.list_plans())
 
-#Example using the 'ImageCollection' API.
-
-datacube = con.imagecollection("S2_L2A_T32TPS_20M")
-datacube = datacube.filter_bbox( west=652000, south=5181000, east=672000, north=5161000, crs="EPSG:32632")
-datacube = datacube.filter_daterange(extent=["2016-01-01T00:00:00Z", "2016-03-10T23:59:59Z"])
-datacube = datacube.ndvi( nir="B04", red="B8A")
+# load a specific dataset
+datacube = ImageCollectionClient.load_collection(session = con, collection_id = "S2_L2A_T32TPS_20M",
```
[the remark this answers is not in the document]
Comment (bgoesswe) [reply examples/eurac_example.py]: fixed in the last commit

File: openeo/internal/filter_bbox.json (hunk 1 above)
Comment (bgoesswe) [reply openeo/internal/filter_bbox.json]: Sry that was just for testing, so should have never bin in the repository:
Removed in the last commit

File: openeo/internal/processes.py (hunk 2 above)
Comment (bgoesswe) [reply openeo/internal/processes.py]: Added a header text with the suggested metadata, also added path to the generation tool.

File: openeo/internal/processes.py (hunk 3 above)
Comment (bgoesswe) [reply openeo/internal/processes.py]: Thanks! I used it to make the comments prettier.

**Comment by soxofaan** [comment]
#153  introduces a 1.0-style variation of the process source code generator

I think we can close this PR #112

**Pull request closed without merging**